integration/spark/src/main/scala/org/apache/spark/sql/secondaryindex/events/SILoadEventListenerForFailedSegments.scala

```diff
@@ -78,20 +78,29 @@ class SILoadEventListenerForFailedSegments extends OperationEventListener with L
                   .getTableMetadata(TableIdentifier(indexTableName,
                     Some(carbonLoadModel.getDatabaseName))).storage.properties
                   .getOrElse("isSITableEnabled", "true").toBoolean
+                val indexTable = metaStore
+                  .lookupRelation(Some(carbonLoadModel.getDatabaseName), indexTableName)(
+                    sparkSession)
+                  .asInstanceOf[CarbonRelation]
+                  .carbonTable
 
-                if (!isLoadSIForFailedSegments) {
+                val metaPathMainTbl: String = carbonTable.getMetadataPath
+                val listOfLoadFolderDetailsArrayMainTbl: Array[LoadMetadataDetails] =
```

Comment:
Handle Same as above comment

Reply:
@Indhumathi27  Here we are preparing list of metadata path of main table and si table. So adding a new method for just getting the metadata path/list of metadatapaths is no use in every case.